src/v/cluster/partition_leaders_table.h

```diff
@@ -193,27 +187,45 @@ class partition_leaders_table {
         model::revision_id partition_revision;
     };
 
-    std::optional<leader_meta>
+    using partition_leaders = absl::btree_map<model::partition_id, leader_meta>;
```

Comment:
Just looked at a profile again with this branch and perf looks much improved. 

I tested using `node_hash_map` instead of the `btree_map` for the inner map and this further improves things quite measurably by 2x in my 9 topics at 5k partitions each test.

Did you chose btree_map intentionally here for some reason?